source/dol/bases/d_CourseSelectGuide.cpp

```diff
@@ -0,0 +1,1000 @@
+    }
+}
+
+void dCourseSelectGuide_c::draw() {
+    if (mInitialized && mEnabled) {
+        mLayout.entry();
+    }
+}
+
+void dCourseSelectGuide_c::doDelete() {
+    mLayout.doDelete();
+}
+
+void dCourseSelectGuide_c::CourseSelectSet(dWmLib::CourseType_e type) {
+    static const int sc_startPointIcons[] = {
+        MSG_CS_ICON_START_RIGHT, // W1
+        MSG_CS_ICON_START_UP,    // W2
+        MSG_CS_ICON_START_RIGHT, // W3
+        MSG_CS_ICON_START_UP,    // W4
+        MSG_CS_ICON_START_RIGHT, // W5
+        MSG_CS_ICON_START_UP,    // W6
+        MSG_CS_ICON_START_RIGHT, // W7
+        MSG_CS_ICON_START_UP,    // W8
+        MSG_CS_ICON_START_UP,    // W9
+    };
+    dInfo_c *info = dInfo_c::m_instance;
+    MsgRes_c *msgRes = dMessage_c::getMesRes();
+    int actualWNo = mWorldNo + 1;
+    if (type == 1) {
+        mCourseNo = -2;
+        return;
+    }
+    if (type == 3 || type == 4) {
+        type = (dWmLib::CourseType_e) (mCourseNo + 1);
+    } else if (type == 2) {
+        mCourseType = 11;
+    } else {
+        mCourseType = 5;
+    }
+    info->field_3b4 = actualWNo;
+    mpTextBoxes[T_worldNum_00]->setMessage(msgRes, BMG_CATEGORY_COURSE_SELECT_GUIDE, MSG_CS_CURR_WORLD, 0);
+    mRestAlphaTarget = 255;
+    int messageID;
+    switch (mCourseType) {
+        default:
+            info->field_3b8 = type;
+            mpTextBoxes[T_cSelect_00]->setMessage(msgRes, BMG_CATEGORY_COURSE_SELECT_GUIDE, MSG_CS_COURSE_NUM, 0);
+            mpTextBoxes[T_cSelect_00]->setVisible(true);
+            mpTextBoxes[T_cSelect_pic]->setVisible(false);
+            return;
+        case 1: messageID = MSG_CS_ICON_GHOST_HOUSE; break;
+        case 2: messageID = MSG_CS_ICON_TOWER; break;
+        case 3:
+            if (mWorldNo != WORLD_8) {
+                messageID = MSG_CS_ICON_CASTLE;
+            } else {
+                messageID = MSG_CS_ICON_W8_CASTLE;
+            }
+            break;
+        case 4:
+            if (dScWMap_c::IsCourseType(mWorldNo, mCourseNo, dScWMap_c::COURSE_TYPE_KINOKO_HOUSE_1UP)) {
+                messageID = MSG_CS_ICON_KINOKO_HOUSE_1UP;
+            } else if (dScWMap_c::IsCourseType(mWorldNo, mCourseNo, dScWMap_c::COURSE_TYPE_KINOKO_HOUSE_STAR)) {
+                messageID = MSG_CS_ICON_KINOKO_HOUSE_STAR;
+            } else {
+                messageID = MSG_CS_ICON_KINOKO_HOUSE_RED;
+            }
+            break;
+        case 5: messageID = MSG_CS_ICON_JUNCTION; break;
+        case 6: messageID = MSG_CS_ICON_CANNON; break;
+        case 8:
+            if (dWmLib::isKoopaShipAnchor()) {
+                messageID = MSG_CS_ICON_ANCHOR;
+            } else {
+                messageID = MSG_CS_ICON_AIRSHIP;
+            }
+            break;
+        case 10:
+            mRestAlphaTarget = 0;
+            messageID = MSG_CS_ICON_PEACH_CASTLE;
+            break;
+        case 11:
+            if (dWmLib::getStartPointKinokoHouseKindNum() == 0 || dWmLib::IsCourseClear(mWorldNo, STAGE_RESCUE)) {
```

Comment:
STAGE_RESCUE is not correct, because this is the starting toad house for the current world. However feel free to keep it like this for now if you want.